tasks/sign-base64-blob/tests/test-sign-base64-blob.yaml

```diff
@@ -0,0 +1,117 @@
+  workspaces:
+    - name: tests-workspace
+  tasks:
+    - name: setup
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      taskSpec:
+        workspaces:
+          - name: data
+        steps:
+          - name: setup-values
+            image: quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+
+              cat > $(workspaces.data.path)/data.json << EOF
+              {
+                "sign": {
+                  "request": "blob-signing-pipeline",
+                  "configMapName": "signing-config-map"
+                }
+              }
+              EOF
+
+              mkdir -p $(workspaces.data.path)/binaries
+              touch $(workspaces.data.path)/binaries/foo_SHA256SUMS
+    - name: run-task
+      taskRef:
+        name: sign-base64-blob
+      params:
+        - name: requester
+          value: testuser
+        - name: blob
+          value: test-blob
+        - name: binariesPath
+          value: binaries
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      runAfter:
+        - setup
+    - name: check-result
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      taskSpec:
+        steps:
+          - name: check-result
+            image: quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+
+              internalRequest="$(kubectl get internalrequest --sort-by=.metadata.creationTimestamp --no-headers)"
+              params=$(kubectl get internalrequest ${internalRequest} -o jsonpath="{.spec.params}")
+
+              if [ $(jq -r '.blob' <<< "${params}") != "test-blob" ]; then
+                echo "blob does not match"
+                exit 1
+              fi
+
+              if [ $(jq -r '.config_map_name' <<< "${params}") != "signing-config-map" ]
+              then
+                echo "config_map_name does not match"
+                exit 1
+              fi
+
+              if [ $(jq -r '.requester' <<< "${params}") != "testuser" ]
+              then
+                echo "requester does not match"
+                exit 1
+              fi
+
+              if [ $(jq -r '.pipeline_image' <<< "${params}") != \
+                 "quay.io/redhat-isv/operator-pipelines-images:released" ]
+              then
+                echo "pipeline_image does not match"
+                exit 1
+              fi
+
+              binaries_path=$(workspaces.data.path)/binaries
+              created_file=$(ls $binaries_path | grep sig)
+              if [ $created_file != "foo_SHA256SUMS.sig" ]
+              then
+                echo "Unexpected filename for the signed file"
+                exit 1
+              fi
+
+              file_content=$(cat $binaries_path/foo_SHA256SUMS.sig)
```

Comment:
Here I would suggest instead to add a condition to the `kubectl` mock to check that the call is what you expect and then echo a dummy signature payload and check that it's in the .sig file here.

Reply:
Done!